setup.py

```diff
@@ -114,20 +112,22 @@ def recursive_data_files(directory, install_directory):
     "macaroonbakery",
     "mypy-extensions",
     "overrides",
+    "packaging",
     "progressbar",
     "pyelftools",
     # Pygit2 and libgit2 need to match versions.
     # Further info: https://www.pygit2.org/install.html#version-numbers
     "pygit2~=1.13.0",
+    "pylxd; sys_platform == 'linux'",
     "pymacaroons",
+    "python-apt @ https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.4.0ubuntu1/python-apt_2.4.0ubuntu1.tar.xz ; sys_platform == 'linux'",
```

Comment:
@lengau - Question for your packaging expertise (and I recall you touching this dep in another repo this year).

This is only going to break developers using non-ubuntu distros trying to install snapcraft from source?

I think the solution will be to require using the snapcraft dev container (or the new rock).

Reply:
Yeah this one's a weird one. I can see the following effects:

1. Prevents publishing on pypi (I don't think that's relevant since we haven't published snapcraft there since the 4.x branch.)
2. Non-apt-based systems won't be able to build snapcraft without modifying this file.
3. Ubuntu systems with an incompatible version of apt won't be able to build snapcraft without modifying this file to provide their own source tarball (already true)

I think the best strategy for our `setup.py` and `pyproject.toml` files is probably what [craft-application](https://github.com/canonical/craft-application/blob/main/pyproject.toml#L63-L65) and [craft-parts](https://github.com/canonical/craft-parts/blob/aa0416efcdc68af34fdaf60263c8a48c67e39458/setup.py#L109) have.

The reason for requiring `>=2.0` is a workaround for the fact that there are old versions of python-apt published on pypi. This forces the user to provide another source for the package. In Rockcraft and Charmcraft we do that through a [separate requirements file](https://github.com/canonical/rockcraft/blob/main/requirements-jammy.txt) for common systems, so that might be the best way to do it here too.

Unrelated, but there is also a newer source package for jammy: https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/python-apt/2.4.0ubuntu2/python-apt_2.4.0ubuntu2.tar.xz

Reply:
After a bit further thought, I think this is fine. It doesn't provide any regressions in behaviour, but it did bring up something I should've fixed. I made an issue for it here: https://github.com/snapcore/snapcraft/issues/4496